src/main/kotlin/no/item/xp/plugin/GenerateTypeScriptTask.kt

```diff
@@ -1,27 +1,96 @@
 package no.item.xp.plugin
 
+import arrow.core.Either
 import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
 import javax.inject.Inject
+import javax.xml.parsers.DocumentBuilderFactory
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.XmlFile
+import no.item.xp.plugin.models.XmlType
+import no.item.xp.plugin.parser.parse
+import no.item.xp.plugin.util.GenerateInterfaceName
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.Document
+import java.lang.Exception
+import javax.xml.parsers.DocumentBuilder
 
+@Suppress("UNUSED_VARIABLE")
 open class GenerateTypeScriptTask @Inject constructor(private val extension: GenerateTypeScriptExtension) : DefaultTask() {
 
+  private val docBuilder: DocumentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+
     @TaskAction
     fun generateTypeScript() {
-      val files = findXmlFiles()
-      println("Found XML files: ")
-      println("--------------------")
-      files.forEach { println(it) }
-      // 2.send dem til en parser
-      // 3.send resultatet til en writer
-      // 4.exit
-    }
-
+      val files: Sequence<String> = findXmlFiles()
+      files.forEach { parseFile(it) }
+  }
   private fun findXmlFiles(): Sequence<String> {
     return File(extension.fileDir)
       .walk()
-      .filter { it.isFile && it.extension == "xml" }
+      .filter { it.isFile }
       .map { it.absolutePath }
   }
+
+  private fun parseFile(filePath: String) {
+    val xmlType: Either<Throwable, XmlType> = getXmlType(filePath)
+    var xmlTypeValue: XmlType? = null
+    when(xmlType){
+      is Either.Right -> {
+        xmlTypeValue = xmlType.b
+      }
+    }
+    if(xmlTypeValue!=null) {
+      val xmlFile: XmlFile = generateXmlFile(filePath, xmlTypeValue)
+      val document: Document = readXml(xmlFile)
+      parse(document)
+
+      val interfaceName: String? = GenerateInterfaceName(File(filePath))
+      val xml: Either<Throwable, Sequence<GeneratedField>> = parse(document)
+      /*if (interfaceName != null) {
+      TypeScriptWriter(interfaceName, xmlType, xml)
+    }*/
+    }
+  }
+
+  private fun getXmlType(filePath: String): Either<Throwable, XmlType> {
+    val child: Path = Paths.get(filePath).toAbsolutePath()
+    return returnIsChild(child)
+  }
+
+  private fun returnIsChild(child: Path): Either<Throwable, XmlType> {
+    for (pair: Pair<String, XmlType> in pairList) {
+      if (isChild(child, pair.first)) {
+        return Either.right(pair.second)
+      }
+    }
+    return Either.left(Exception("could not find XmlType for $child"))
+  }
```

Comment:
Try to avoid using `for`-loops, and if possible, if possible try to use a function that encapsulates both `left` and `right` at once, instead of manually applying them.
```suggestion
  private fun returnIsChild(child: Path): Either<Throwable, XmlType> =
    pairList
      .find { isChild(child, it.first) }
      .map { it.second }
      .toEither { Exception("could not find XmlType for $child") }
```
If you look at my suggestion, you get more of a "data transformation pipeline" feel.